### DISCUSSION TITLE
ImageMagick Support Heic Image Encode ?

### DISCUSSION THREAD

**Discussion by wsx958191** [ImageMagick/ImageMagick#5540] [Help]
hi:
   When I was testing HEIC code, ImageMagick had been compiled into -- with HEIC = yes. I checked that all the modules commissioned had been successfully pushed. I found the following errors by using the command test. Except for heif format, other formats were verified normally
   
<img width="774" alt="企业微信截图_bc8a4e94-2da4-4395-8fec-5b4c69d455e2" src="https://user-images.githubusercontent.com/66000167/189018479-c722e882-983a-4396-8f97-6fecc8ebec9a.png">
<img width="742" alt="企业微信截图_8fa779bf-3daf-4c1a-9ff6-93c471ab8d5f" src="https://user-images.githubusercontent.com/66000167/189018568-0fcfb6ce-ef07-4f54-8791-0ca3c1c1c68b.png">

**Comment by snibgo** [comment]
What does "`mogrify -list format`" say? Perhaps:
```
     HEIC  r--   High Efficiency Image Format (1.12.0)
     HEIF  r--   High Efficiency Image Format (1.12.0)
```
"r--" means this version of IM can read but not write those formats.

**Comment by urban-warrior** [comment]
Writing HEIC is conditional upon which compression options are supported by your release of the HEIC library.  For example, you need HEVC compression supported by the HEIC delegate library to write HEIC images.  We have that so `identify -list format` returns:

```
     HEIC  HEIC      rw+   High Efficiency Image Format (1.12.0)
     HEIF  HEIC      rw+   High Efficiency Image Format (1.12.0)
```
for us where the `w` means that writing the HEIC/HEIF format is supported in our instance of ImageMagick.
